### DISCUSSION TITLE
TeslaMate keep car awake under certain conditions

### DISCUSSION THREAD

**Discussion by tbarnekov** [teslamate-org/teslamate#1449] [Q&A] [answered]
I have a 2015 Model S and since I started using TeslaMate about a month ago, I've noticed a pattern where the car has trouble sleeping. At first it seemed a little random, but I think I've figured out what's going on...

Typically the car will reliably sleep within 20-30 minutes after a drive. It will stay asleep until it's woken up by a keyfob being present or if it has to call back to the mothership or whatever. It can sleep for hours - no problem.

But if the car is woken up and **not** being driven, it'll go into insomniac mode. It'll be awake for 20-30 minutes and go to sleep for seconds to minutes an then wake up again.

![image](https://user-images.githubusercontent.com/37448419/111286625-af2f2c80-8642-11eb-9733-3c5cc2307a6c.png)

When examining the periods of insomnia, I'll often (but not always) see the following pattern in the logs:

```
2021-03-16 07:50:56.560 car_id=1 [info] Fetching vehicle state ...
2021-03-16 07:50:57.087 car_id=1 [info] Start / :asleep
2021-03-16 07:50:57.092 car_id=1 [info] Disconnecting ...
2021-03-16 07:51:46.694 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/150463753864/vehicle_data -> 408 (9233.707 ms)
2021-03-16 07:51:46.694 car_id=1 [error] Error / :vehicle_unavailable
2021-03-16 07:52:19.120 car_id=1 [info] Start / :online
2021-03-16 07:52:19.128 car_id=1 [info] Connecting ...
```

The /vehicle_data request always seem to take about 9.2 to 9.3 seconds so that tells me that TeslaMate is querying the /vehicle_data endpoint **after** the vehicle has gone to sleep. In the example above TeslaMate has started the query about 40 seconds after the car reported going to sleep.

The /vehicle_data endpoint is known to wake up the car and I think that's what's causing the pattern I'm seeing.

Previously I've noticed that the 408 Vehicle Unavailable message is often logged around 20 seconds after Start / :asleep but this seems to have changed after upgrading to 1.21.6. Now it happens within a minute of going to sleep. I don't remember which versions I've tried before the latest upgrade, but the pattern seems to have been there on every version I've tried.

**Expected behavior**

I expect the car to go to sleep and stay asleep just as it does after finishing a drive.

**Environment**

- TeslaMate version: 1.21.6
- Type of installation: Docker
- OS TeslaMate is installed on: Ubuntu

**Comment by adriankumpf** [comment, marked as answer]
> The /vehicle_data endpoint is known to wake up the car and I think that's what's causing the pattern I'm seeing.

This is not true. These requests cannot wake up the car. 

They will prevent the car from _going into sleep mode_ when called. For this reason TeslaMate stops polling the car after thee minutes (`[info] Suspending logging`) until it is asleep. Your car waking up is caused by something else. Check out the other threads with tips for this issue, for instance: https://github.com/adriankumpf/teslamate/issues/571#issuecomment-611987746

This seems to be particularly relevant:

> If you have a Model S or X built prior to March 2018 make sure 'Always Connected' is turned off and 'Energy Savings' is turned on in the vehicle.

**Reply by bahree** [reply]
So, I started seeing this issue in the last 2-3 days as well with no changes that I can attribute to this behavior. Here is my snapshot from the last 24 hours as an example. I am hoping I can see if there is anything obvious.

![image](https://user-images.githubusercontent.com/3529468/127376006-eac3656a-5ac7-44fc-97d7-982f15df81af.png)

**Reply by tbarnekov** [reply]
That may be bad news. Not long after I posted this, I got a ‘Charge level reduced’ warning on the dashboard. The car wouldn’t charge past 38% after that. 

I ended up getting the main drive battery replaced under warranty, and after an initial calibration period the car has slept much more consistently.

I used to see vampire drain upwards of 150 W/h which has now dropped to 10-25 W/h after the battery replacement.
